savant/client/runner/source.py

```diff
@@ -107,37 +109,29 @@ def __del__(self):
         logger.info('Terminating ZeroMQ connection')
         self._writer.shutdown()
 
-    def __call__(self, source: Frame, send_eos: bool = True) -> SourceResult:
-        """Send a single frame to ZeroMQ socket.
-
-        :param source: Source of the frame to send. Can be an instance
-            of FrameSource or a tuple of VideoFrame and content.
-        :param send_eos: Whether to send EOS after sending the frame.
-        :return: Result of sending the frame.
-        """
+    def __call__(self, source: Source, send_eos: bool = True) -> SourceResult:
+        """Handle the callable interface for this object. For more information, see the send method."""
 
         return self.send(source, send_eos)
 
-    def send(self, source: Frame, send_eos: bool = True) -> SourceResult:
-        """Send a single frame to ZeroMQ socket.
+    def send(self, source: Source, send_eos: bool = True) -> SourceResult:
+        """Send source data to ZeroMQ socket.
 
         :param source: Source of the frame to send. Can be an instance
-            of FrameSource or a tuple of VideoFrame and content.
-        :param send_eos: Whether to send EOS after sending the frame.
-        :return: Result of sending the frame.
+            of FrameSource or a tuple of VideoFrame and content, or a batch of frames with source ID.
+        :param send_eos: Whether to send EOS after sending the source.
+        :return: Result of sending the source.
         """
 
         if self._health_check is not None:
             self._health_check.wait_module_is_ready()
 
-        zmq_topic, message, content, result = self._prepare_video_frame(source)
-        logger.debug(
-            'Sending video frame %s/%s.',
-            result.source_id,
-            result.pts,
-        )
-        self._send_zmq_message(zmq_topic, message, content)
-        logger.debug('Sent video frame %s/%s.', result.source_id, result.pts)
+        if isinstance(source, tuple) and isinstance(source[0], VideoFrameBatch):
+            batch, source_id = source
+            result = self._send_batch(source_id, batch)
+        else:
+            result = self._send_frame(source)
+
         if send_eos:
             self.send_eos(result.source_id)
```

Comment:
Batch can contain frames from different sources. E.g. when the module needs to process video from 2 cameras synchronously. Methods `send` and `send_iter` should send EOS for each `source_id`. I suggest adding `source_ids` to `SourceResult` and using it.